.github/workflows/ci.yml

```diff
@@ -0,0 +1,93 @@
+name: CI
+
+on:
+  push:
+    branches: [ main ]
+  release:
+    types: [published]
+    branches: [ main ]
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  build:
+    name: Build and test
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-22.04, windows-2022 ]
+        include:
+          - os: ubuntu-22.04
+            platform: linux
+          - os: windows-2022
+            platform: windows
+
+    env:
+      VCPKG_ROOT: 'vcpkg'
+      VCPKG_BINARY_SOURCES: 'clear;nuget,GitHub,readwrite'
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Set up MSVC (Windows)
+      if: matrix.os == 'windows-2022'
+      uses: ilammy/msvc-dev-cmd@v1
+      with:
+          arch: x64
+
+    - name: Install dependencies (Linux)
+      if: matrix.os == 'ubuntu-22.04'
+      run: |
+        sudo apt update
+        sudo apt upgrade -y
+        sudo apt install --no-install-recommends build-essential libtbb-dev ninja-build doxygen graphviz
+
+    - name: Install VCPKG
+      run: |
+        git clone https://github.com/Microsoft/vcpkg.git
+        cd vcpkg
+        ./bootstrap-vcpkg.sh -disableMetrics
+
+    - name: Build HealthGPS (debug)
+      run: |
+        cmake --preset=${{ matrix.platform }}-debug -DBUILD_BENCHMARKS=ON
+        cmake --build --preset=debug-build-${{ matrix.platform }} --target=install 
+
+    - name: Build HealthGPS (release)
+      run: |
+        cmake --preset=${{ matrix.platform }}-release -DBUILD_BENCHMARKS=ON
+        cmake --build --preset=release-build-${{ matrix.platform }} --target=install 
+
+    - name: Upload artifacts
+      uses: actions/upload-artifact@v3
+      with:
+        name: health-gps-${{ matrix.platform }}
+        path: |
+          ${{ github.workspace }}/out/install/${{ matrix.platform }}-*/*
+
+    - name: Run tests
+      run: ctest --preset=core-test-${{ matrix.platform }}
+
+    - name: Zip output folder
+      if: startsWith(github.ref, 'refs/tags/')
+      working-directory: ${{github.workspace}}/out/install/${{ matrix.platform }}-release/bin/
+      run: |
+        mkdir ${{github.workspace}}/artifact
+        7z a -tzip ${{github.workspace}}/artifact/health_gps_${{ matrix.platform }}.zip *.dll *.Console*
+
+    - name: Upload release artifacts
+      if: startsWith(github.ref, 'refs/tags/')
+      uses: softprops/action-gh-release@v1
+      with:
+          files: artifact/health_gps_${{ matrix.platform }}.zip
+
+    - name: Doxygen release API deploy
```

Comment:
If creating the documentation does not require any of the previous steps - except fetching the repo - i would separate it into a different job, so it can run in parallel with the testing ones.

Reply:
The thing is, it really shouldn't, except that they've integrated building the docs with CMake, so it does kinda have to come after invoking CMake. Really though you should just be able to run `doxygen` manually though to just get the docs (it's fine to *also* have CMake optionally build the docs if the user wants, but it's annoying to have it as a hard dependency).

They're definitely not alone in doing this, but it probably makes sense to break that dependency. Again, maybe I should open an issue and make a new PR for it?